src/loki/search.cc

```diff
@@ -642,16 +650,20 @@ struct bin_handler_t {
           // the last one wasnt in the radius so replace it with this one because its better or is
           // in the radius
           if (!last_in_radius) {
+            if (closer_external_reachable)
+              p_itr->closest_external_reachable = batch->back().sq_distance;
             batch->back() = std::move(*c_itr);
             // last one is in the radius but this one is better so put it on the end
           } else if (better) {
             batch->emplace_back(std::move(*c_itr));
-            // last one is in the radius and this one is not as good so put it on before it
+            // last one and this one are both in the radius but this one is not as good
           } else {
             batch->emplace_back(std::move(*c_itr));
             std::swap(*(batch->end() - 1), *(batch->end() - 2));
           }
-        }
+        } // not in radius or better and reachable and closer than closest one outside of radius
+        else if (closer_external_reachable)
+          p_itr->closest_external_reachable = c_itr->sq_distance;
```

Comment:
same thing, remember how close this rejected one was for later so we can use it to cull low reachability candidates